src/main/java/mate/academy/impl/AbstractDao.java

```diff
@@ -0,0 +1,99 @@
+package mate.academy.impl;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public abstract class AbstractDao<T, I extends Serializable> {
```

Comment:
you already have AbstractDao, RoleDaoImpl, UserDaoImpl in the mate.academy.dao.impl. Why did you add it?

Reply:
I accidentally deleted these classes and couldn't restore them, so I created new ones

Reply:
@anastasiavasylyk I don't see any commit changes where you are removing the existing class. If you will merge this PR in production - you will have two AbstractDao classes. One in `mate.academy.dao.impl` and the other in `mate.academy.impl`. So please remove this code from this PR.